src/LogicalCalculator.tsx

```diff
@@ -0,0 +1,206 @@
+import React, { useState } from 'react';
+import { DataProps } from './App';
+
+
+const LogicalCalculator:React.FC<DataProps> = ({ data }) => {
+    const [operation, setOperation] = useState<string>("MIN");
+    const [startColumnType, setStartColumnType] = useState<string>("battery_power");
+    const [startIndex, setStartIndex] = useState<number>(1);
+    const [endColumnType, setEndColumnType] = useState<string>("battery_power");
+    const [endIndex, setEndIndex] = useState<number>(1);
+    const [rankPointType, setRankPointType] = useState<string>("battery_power");
+    const [rankPointIndex, setRankPointIndex] = useState<number>(1);
+    const [result, setResult] = useState<number | undefined>(undefined);
+
+    let rangeArray: number[] = [];
+
+    const handleOperationChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
+        setOperation(e.target.value);
+    };
+
+    const handleStartColumnTypeChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
+        setStartColumnType(e.target.value);
+    }
+
+    const handleStartIndexChange = (e: any) => {
+        setStartIndex(e.target.value);
+    }
+
+    const handleEndColumnTypeChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
+        setEndColumnType(e.target.value);
+    }
+
+    const handleEndIndexChange = (e: any) => {
+        setEndIndex(e.target.value);
+    }
+
+    const handleRankPointTypeChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
+        setRankPointType(e.target.value);
+    }
+
+    const handleRankPointIndexChange = (e: any) => {
+        setRankPointIndex(e.target.value);
+    }
+
+    const minValue = Math.min(startIndex, endIndex);
+    const maxValue = Math.max(startIndex, endIndex);
+
+    if ((startColumnType === "battery_power" && endColumnType === "ram") || (startColumnType === "ram" && endColumnType === "battery_power")) {
```

Comment:
Would be better to make a separated type for column types rather than just texts.
COLUMNTYPE.BATTERYPOWER
COLUMNTYPE.RAM
COLUMNTYPE.PXHEIGHT